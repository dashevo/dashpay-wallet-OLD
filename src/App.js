import React from 'react';
import { StoreProvider, LanguageProvider } from 'containers';
import { ThemeProvider } from 'hooks/Theme';
import { Navigator } from 'navigations';
import translations from 'translations';
import themes from 'themes';
import state from 'state';

import DapiPoll from 'components/DapiPoll';

function App(props) {
  return (
    <StoreProvider store={state}>
      <DapiPoll />
      <LanguageProvider translations={translations}>
        <ThemeProvider themes={themes}>
          <Navigator {...props} />
        </ThemeProvider>
      </LanguageProvider>
    </StoreProvider>
  );
}

export default App;
