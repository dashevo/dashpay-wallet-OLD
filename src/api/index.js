/**
 * Copyright (c) 2014-present, Dash Core Group, Inc.
 */
'use strict';

export * from './middleware';
