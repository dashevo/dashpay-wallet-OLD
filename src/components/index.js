/**
 * Copyright (c) 2014-present, Dash Core Group, Inc.
 *
 * @flow
 */
export { default as Icon } from './Icon';

// TODO: "render props" + global variables for styles
export { default as Avatar } from './Avatar';
export { default as Logo } from './Logo';
export { default as ProgressBar } from './ProgressBar';
export { default as SharedElement } from './SharedElement';
export { default as ThemedButton } from './ThemedButton';
export { default as LabeledSwitch } from './LabeledSwitch';
export { default as SelectableList } from './SelectableList';
export { default as IconButton } from './IconButton';
export { default as RecipientInput } from './RecipientInput';
export { default as CopyAddressButton } from './CopyAddressButton';
export { default as NavBar } from './NavBar';
export { default as SwipeableRow } from './SwipeableRow';
export { default as FormattedText } from './FormattedText';
export { default as FormattedDate } from './FormattedDate';
export { default as FormattedTime } from './FormattedTime';
export { default as FormattedRelative } from './FormattedRelative';
export { default as FormattedNumber } from './FormattedNumber';
export * from './Animation';
