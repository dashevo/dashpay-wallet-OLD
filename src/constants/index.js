/**
 * Copyright (c) 2014-present, Dash Core Group, Inc.
 *
 * @flow
 */
import { Dimensions } from "react-native";

export { default as COLORS } from "./colors";
export { default as THEMES } from "./theming";
export { default as LENGTHS } from "./lengths";
export { default as ALTERNATIVE_CURRENCIES } from "./alternativeCurrencies";
export { default as TXTYPES } from "./txtypes";

export * from "./defaults";

export const { height: SCREEN_HEIGHT, width: SCREEN_WIDTH } = Dimensions.get(
  "window"
);
