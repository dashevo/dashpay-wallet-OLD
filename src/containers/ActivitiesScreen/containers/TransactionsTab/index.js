/**
 * Copyright (c) 2014-present, Dash Core Group, Inc.
 *
 * @flow
 */

// External dependencies
// External dependencies
import * as React from 'react';
import { SectionList } from 'react-native';
import { connect } from 'react-redux';

// Internal dependencies
import View from 'components/View';
import Text from 'components/Text';
import selectors from './selectors';
import styles from './styles';

type Props = {};
type State = {};

class TransactionsTab extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    (this: any).handleViewableItemsChanged = this.handleViewableItemsChanged.bind(
      this
    );
    (this: any).renderSectionHeader = this.renderSectionHeader.bind(this);
    (this: any).keyExtractor = this.keyExtractor.bind(this);
    (this: any).renderItem = this.renderItem.bind(this);

    this.state = {};
  }

  keyExtractor(item, index) {
    return item + index;
  }

  handleViewableItemsChanged(info) {}

  renderSectionHeader(info) {
    return (
      <View style={styles.titleWrapper}>
        <Text style={styles.title}>{info.section.title}</Text>
      </View>
    );
  }

  renderItem(props) {
    return <Text>{props.item.name}</Text>;
  }

  ListHeaderComponent = () => {
    return <Text>{'ListHeaderComponent'}</Text>;
  };

  render() {
    const { transactions } = this.props;
    const sections = [
      {
        title: 'Title1',
        data: transactions
      }
    ];
    return (
      <SectionList
        ListHeaderComponent={this.ListHeaderComponent}
        onViewableItemsChanged={this.handleViewableItemsChanged}
        renderSectionHeader={this.renderSectionHeader}
        stickySectionHeadersEnabled={true}
        keyExtractor={this.keyExtractor}
        renderItem={this.renderItem}
        sections={sections}
        style={styles.container}
      />
    );
  }
}

export default connect(selectors)(TransactionsTab);
