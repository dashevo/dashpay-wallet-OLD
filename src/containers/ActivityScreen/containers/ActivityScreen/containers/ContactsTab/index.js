/**
 * Copyright (c) 2014-present, Dash Core Group, Inc.
 *
 * @flow
 */

// External dependencies
import * as React from 'react';
import { connect } from 'react-redux';

// Internal dependencies
import Transactions from './components/Transactions';
import selector from './selectors';
import actions from './actions';

export default connect(
  selector,
  actions
)(Transactions);
