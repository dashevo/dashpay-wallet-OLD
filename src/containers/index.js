/**
 * Copyright (c) 2014-present, Dash Core Group, Inc.
 *
 * @flow
 */
export * from './SplashScreen';
export * from './HomeScreen';
