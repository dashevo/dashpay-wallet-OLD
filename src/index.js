/**
 * Copyright (c) 2014-present, Dash Core Group, Inc.
 */
export { default as App } from './App';
