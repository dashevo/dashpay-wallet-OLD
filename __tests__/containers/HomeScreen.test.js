import React from 'react';
import { HomeScreen } from '../../src/containers';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
