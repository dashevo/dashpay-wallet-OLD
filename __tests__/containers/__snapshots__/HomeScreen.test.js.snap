// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#088BE2",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/logo.png",
      }
    }
    style={
      Object {
        "height": "20%",
        "resizeMode": "contain",
        "width": "40%",
      }
    }
  />
  <Image
    source={
      Object {
        "uri": "https://api.adorable.io/avatars/285/dashAvatar.png",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "resizeMode": "cover",
        "width": 100,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 24,
        "textAlign": "center",
      }
    }
  >
    HomeScreen
  </Text>
</View>
`;
